snykTask/src/task-args.ts

```diff
@@ -102,31 +111,45 @@ class TaskArgs {
     return this.projectName;
   }
 
+  // validate based on testTypeSeverityThreshold applicable thresholds
   public validate() {
-    if (this.failOnThreshold) {
-      if (this.isNotValidThreshold(this.failOnThreshold)) {
-        const errorMsg = `If set, failOnThreshold must be '${Severity.CRITICAL}' or '${Severity.HIGH}' or '${Severity.MEDIUM}' or '${Severity.LOW}' (case insensitive). If not set, the default is '${Severity.LOW}'.`;
-        throw new Error(errorMsg);
-      }
+    const taskTestType = this.testType || TestType.APPLICATION;
+    const validTestTypes: any = Object.values(TestType);
```

Comment:
I can see that the definition for `TestType` is a a string enum, right?

```
export enum TestType {
  APPLICATION = 'app',
  CODE = 'code',
  CONTAINER_IMAGE = 'container',
}

```
I think we could avoid using `any` here and casting it to:

```ts
    const validTestTypes: string[] = Object.values(TestType);
```

This way, we maintain the benefits of type checking :)

Reply:
corrected